src/rgw/rgw_iam_policy.cc

```diff
@@ -715,24 +716,38 @@ static optional<Principal> parse_principal(CephContext* cct, TokenID t,
       return Principal::tenant(std::move(a->account));
     }
 
-    static const char rx_str[] = "([^/]*)/(.*)";
-    static const regex rx(rx_str, sizeof(rx_str) - 1,
-			  ECMAScript | optimize);
-    smatch match;
-    if (regex_match(a->resource, match, rx)) {
-      ceph_assert(match.size() == 3);
+    typedef boost::tokenizer<boost::char_separator<char> > tokenizer;
+    boost::char_separator<char> sep("/");
+    tokenizer tokens(a->resource, sep);
 
-      if (match[1] == "user") {
+    auto it1 = tokens.begin();
+
+    int size = distance(tokens.begin(), tokens.end());
+    auto it2 = tokens.begin();
+    while (it2 != tokens.end()) {
+        if (distance(tokens.begin(), it2) == (size - 1)) {
```

Comment:
it looks like we're just trying to parse the first and last token when there are multiple `/`s (am i reading this right?). if so, `boost::tokenizer` is probably overkill here (especially if it means you can't iterate backwards from the end). does this version with `boost::string_view` look better?
```c++
  const boost::string_view input{a->resource};

  const auto first = input.find_first_of('/');
  if (first == input.npos) {
    return boost::none;
  }
  const auto last = input.find_last_of('/');
  ceph_assert(last != input.npos); // find_first_of() already found one

  const auto principal_type = input.substr(0, first);
  const auto entity_name = input.substr(last + 1);
```
since `entity_name` is a `boost::string_view`, you'd use `entity_name.to_string()` instead of `std::move(entity_name)` below